auxin_cli/src/state.rs

```diff
@@ -610,46 +468,55 @@ pub async fn make_context(
 	})
 }
 
+/// Bridge between signal-cli state and auxin state
 pub struct StateManager {
-	pub base_dir: String,
+	/// Signal-cli data directory
+	data_dir: PathBuf,
 }
 
 impl StateManager {
-	pub fn new(base_dir: &str) -> Self {
+	/// Create a new state using the supplied state directory
+	pub fn new(state_dir: &str) -> Self {
 		StateManager {
-			base_dir: base_dir.to_string(),
+			data_dir: Path::new(state_dir).join("data"),
 		}
 	}
-	pub fn get_protocol_store_path(&self, context: &AuxinContext) -> String {
-		let our_id: String = context.identity.address.get_phone_number().unwrap().clone();
+
+	/// Used in tests
+	#[cfg(test)]
+	fn new_test(data_dir: PathBuf) -> Self {
+		StateManager { data_dir }
+	}
+
+	/// Get path to signal-cli protocol state, `<phone_number>.d`
+	fn get_protocol_store_path(&self, context: &AuxinContext) -> PathBuf {
+		// TODO(Diana): Phone numbers, UUIDs, usernames, unwrap.
+		let our_id = context.identity.address.get_phone_number().unwrap();
 		//Figure out some directories.
-		let mut our_path = self.base_dir.clone();
-		our_path.push('/');
-		our_path.push_str(our_id.as_str());
-		our_path.push_str(".d/");
-		our_path
+		self.data_dir.join(our_id).with_extension("d")
 	}
 }
 
+/// Auxin state, compatible with signal-cli
 impl AuxinStateManager for StateManager {
+	/// Load the local identity for `phone_number` from signal-cli
+	///
+	/// In signal_cli's protocol store structure,
+	/// this comes from the file with a name which is your phone number inside the "data" directory.
 	fn load_local_identity(&mut self, phone_number: &E164) -> crate::Result<LocalIdentity> {
-		let user_json = load_signal_cli_user(self.base_dir.as_str(), phone_number)?;
-		let local_identity = local_identity_from_json(&user_json)?;
-		Ok(local_identity)
+		let file = File::open(self.data_dir.join(phone_number))?;
+		from_reader::<_, LocalIdentityJson>(BufReader::new(file))?.try_into()
 	}
+
 	fn load_context(
 		&mut self,
 		credentials: &LocalIdentity,
 		config: AuxinConfig,
 	) -> crate::Result<AuxinContext> {
-		return block_on(make_context(
-			self.base_dir.as_str(),
-			credentials.clone(),
-			config,
-		));
+		// TODO(Diana): Why so much async in this file for a sync interface??
```

Comment:
Many of libsignal-client-rs' protocol store traits use async methods via async-trait, so adding data to them or loading data from them needs an ```.await``` or a ```block_on()```. I believe these are because they're designed to be generic enough to wrap fetching data from the network or a DB.

Reply:
Yeah, but a lot of the internal functions are "async" instead of just at the edges, but most of them don't need to be and it makes it more confusing IMO. Something I want to resolve in the next PR, make the functions sync and add more block_on's, or make them properly async, maybe?